AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/ApkStore.kt

```diff
@@ -84,6 +84,34 @@ internal class ApkStore(
         }
     }
 
+    suspend fun uploadAdditionalApk(filePath: String): UploadedApk {
+        val apkInfo = ApkInfo.create(filePath)
+        getUploadedAdditionalApk(apkInfo)?.let {
+            return it
+        }
+        logger.info {
+            "downloading $filePath"
+        }
+        val bytes = googleCloudApi.downloadAdditionalApk(filePath)
+
+        logger.info {
+            "uploading $filePath"
+        }
+        val relativePath = filePath.split("/").takeLast(3).joinToString("/")
```

Comment:
what is this takeLast(3) is meant for exactly?
ideally, we should NOT be relying on that if possible. (reads very random / hardcoded so if there is a way to make it generic, we should)

e.g. if this is something like:

`gs//<some random bucket in gcp>/some random path`

we should get `some random path` and likely prefix it w/ something else e.g. `additional-apks/` when putting it into our own bucket.